src/Core/Content/Sitemap/Provider/CategoryUrlProvider.php

```diff
@@ -57,17 +63,27 @@ public function getUrls(SalesChannelContext $context, int $limit, ?int $offset =
         if (empty($categories)) {
             return new UrlResult([], null);
         }
+
         $keys = FetchModeHelper::keyPair($categories);
 
-        $seoUrls = $this->getSeoUrls(array_values($keys), 'frontend.navigation.page', $context, $this->connection);
+        // Load categories from the repository to allow decorators and event listeners to modify the result.
```

Comment:
If you want to allow event listeners to modify the result, please dispatch an own event. It will be more performant as the current solution and it is easier for external devs to listen to the standalone events. The dal events are very generic and you have to filter first for the right one which you want to extend.